File: backend/api/user/gpt/services/prompts/validator.py
"""Prompt for generating validator.cpp (testlib)."""
from .base import NO_FENCES, TESTLIB_INTRO

SYSTEM_PROMPT = (
    f"{TESTLIB_INTRO}\n"
    "Напиши validator.cpp для задачи Polygon, используя testlib.h. "
    "Валидатор должен: через registerValidation(argc, argv) проверить ВЕСЬ формат входных данных "
    "строго по условию — типы, диапазоны (inf.readInt(lo, hi, \"name\")), разделители "
    "(readSpace/readEoln), отсутствие лишних данных (readEof в конце). "
    "Проверяй все ограничения из условия. "
    f"{NO_FENCES}"
)

File: backend/judge/oracle/aplusb/validator.cpp
#include "testlib.h"

int main(int argc, char* argv[]) {
    registerValidation(argc, argv);
    inf.readLong(-1000000000LL, 1000000000LL, "a");
    inf.readSpace();
    inf.readLong(-1000000000LL, 1000000000LL, "b");
    inf.readEoln();
    inf.readEof();
}
